ddprof-lib/src/main/cpp/asyncSampleMutex.h

```diff
@@ -0,0 +1,38 @@
+#ifndef ASYNCSAMPLEMUTEX_H
+#define ASYNCSAMPLEMUTEX_H
+
+#include "threadLocalData.h"
+
+// controls access to AGCT
+class AsyncSampleMutex {
+private:
+    ThreadLocalData* _threadLocalData;
+    bool _acquired;
+
+    bool try_acquire() {
+        if (_threadLocalData != nullptr && !_threadLocalData->is_unwinding_Java()) {
+            _threadLocalData->set_unwinding_Java(true);
+            return true;
+        }
+        return false;
+    }
+
+public:
+    AsyncSampleMutex(ThreadLocalData* threadLocalData) : _threadLocalData(threadLocalData) {
+        _acquired = try_acquire();
+    }
+
+    AsyncSampleMutex(AsyncSampleMutex& other) = delete;
```

Comment:
NIT: if we delete this one, we can adjust the other constructors (rule of 5) to ensure we can not copy/move or use = operators.